kuksa-sdk/src/test/kotlin/org/eclipse/kuksa/vssSpecification/VssSpecificationCopyTest.kt

```diff
@@ -25,27 +25,72 @@ import io.kotest.matchers.shouldBe
 import io.kotest.matchers.string.shouldStartWith
 import org.eclipse.kuksa.extension.copy
 import org.eclipse.kuksa.extension.deepCopy
+import org.eclipse.kuksa.extension.not
+import org.eclipse.kuksa.extension.plus
 import org.eclipse.kuksa.proto.v1.Types
 import org.eclipse.kuksa.proto.v1.Types.BoolArray
 import org.eclipse.kuksa.test.kotest.Unit
+import org.eclipse.kuksa.vsscore.model.VssProperty
 
 class VssSpecificationCopyTest : BehaviorSpec({
     tags(Unit)
 
     given("A specification") {
-        val driver = VssDriver()
-        val heartRate = driver.heartRate
+        val vehicle = VssVehicle()
+        var driverHeartRate: VssProperty<Int> = vehicle.driver.heartRate
 
         and("a changed heritage line") {
             val newValue = 70
-            val updatedHeartRate = VssHeartRate(value = newValue)
-            val changedHeritageLine = listOf(updatedHeartRate)
+            val updatedHeartRate = VssDriver.VssHeartRate(value = newValue)
 
             `when`("a deep copy is done with a changed heritage line") {
-                val deepCopiedSpecification = driver.deepCopy(changedHeritageLine)
+                val deepCopiedSpecification = vehicle.deepCopy(0, updatedHeartRate)
 
                 then("it should return the new children as a copy") {
-                    val heartRateValue = deepCopiedSpecification.heartRate.value
+                    val heartRateValue = deepCopiedSpecification.driver.heartRate.value
+
+                    heartRateValue shouldBe newValue
+                }
+            }
```

Comment:
I don't really get the finding.  I am doing a deep copy with a far away heir (HeartRate) and afterwards I check if the HeartRate was added to the deep copy of the Vehicle.

Reply:
This depends on what you want to test. If you want to test the aspect, that you simply "inject" your model into the existing model then yes, your test works. 

If you however want to test if "deepCopy" really returns a deep copy, than it does not work this way, because right now you still did not verify the referential (in)equality of both objects, so basically a change to one object in your "toBeCopied" instance has an impact on an object inside your copied instance.

A deep copy obviously means that all instances between a "toBeCopied" instance and it's copied instance should point to a different object in memory, if it's a shallow copy or deep copy obvious depends on how a list is copied: a shallow copy means, that simply a new list object is used (while all elements are simply re-used) However a deep copy means that a new list is returned with each element inside the list being a new copy as well. referential equality can be checked with "===". (https://kotlinlang.org/docs/equality.html#referential-equality)

I don't see this reflected here, therefore I was asking if you test the deepCopy aspect (because it's explicitely mentioned) or if you just wanted to test the injection of your object inside the "toBeCopied" object

Reply:
The initial idea was the last one but let me add the pointer equals too